### ISSUE TITLE
Can't build vigil 1.22.2 with XMPP feature

### ISSUE THREAD

**Issue by kauron** [valeriansaliou/vigil#108]
Hi there! I'm a happy user of vigil, and I found a problem while trying to build the latest version (1.22.2).

I've tried all these variations to build vigil, and all fail:
```
# 1. From source (README), including all features
cargo build --release --all-features
# 2. From crates.io (README), including all features
cargo install vigil-server --all-features
# 3. Locking dependencies to their current state, including all features
cargo fetch --locked
cargo build --frozen --release --all-features
```
The build fails when trying to compile `libstrophe-0.16.0`, with
```
error[E0433]: failed to resolve: use of undeclared crate or module `log`
   --> $HOME/.cargo/registry/src/github.com-1ecc6299db9ec823/libstrophe-0.16.0/src/connection.rs:522.6
    |
522 |       if log::log_enabled!(log::Level::Debug) {
    |          ^^^ use of undeclared crate or module `log`
```
I'm not too familiar with rust, but I understand that this is `libstrophe`'s fault. Still, I wanted to report the issue here, because users of the XMPP notifier can't build the latest version.

Thanks for creating this project, it's been very useful to me.

**Comment by valeriansaliou** [comment]
Ah snap! Yes, this looks like a libstrophe issue. Can you report it to them? Does it build w/ Rust 1.21 and lower?

**Comment by kauron** [comment]
Yes, I'll report this issue to them. I'm currently building with Rust 1.58, I'll look into building with <=1.21, but it seems unlikely (it's now 4 years old).

**Comment by twistedfall** [comment]
It's fixed in libstrophe version 0.16.1
